File: lib/admin/features/settings/services/platform_settings_service.dart
import 'package:cloud_firestore/cloud_firestore.dart' as fs;

import '../../../../core/services/firebase_service.dart';
import '../../../core/admin_audit_service.dart';
import '../models/platform_settings.dart';

/// `platformSettings/config` is a singleton — one document, one id, always
/// read/written by that fixed reference.
class PlatformSettingsService {
  PlatformSettingsService._();
  static final PlatformSettingsService instance = PlatformSettingsService._();

  fs.DocumentReference<Map<String, dynamic>> get _doc =>
      FirebaseService.instance.firestore.collection('platformSettings').doc('config');

  Future<PlatformSettings> fetch() async {
    final snap = await _doc.get();
    if (!snap.exists || snap.data() == null) return const PlatformSettings();
    return PlatformSettings.fromMap(snap.data()!);
  }

  Future<void> save(PlatformSettings settings, {required String adminId}) async {
    final updated = settings.copyWith(updatedByAdminId: adminId);
    await _doc.set(updated.toMap());
    await AdminAuditService.instance.log(
      action: 'platform_settings.update',
      targetType: 'platformSettings',
      targetId: 'config',
      details: {
        'maintenanceMode': updated.maintenanceMode,
        'minSupportedAppVersion': updated.minSupportedAppVersion,
      },
    );
  }
}

File: admin-web/src/services/platformSettingsService.ts
import { doc, getDoc, setDoc, serverTimestamp } from 'firebase/firestore'
import { db } from '@/lib/firebase'
import { logAuditEvent } from './adminAuditService'
import { mapPlatformSettings, defaultPlatformSettings, type PlatformSettings } from '@/types/models'

/** platformSettings/config is a singleton — one document, one id, always read/written by that fixed reference. */
const settingsDoc = () => doc(db, 'platformSettings', 'config')

export async function fetchPlatformSettings(): Promise<PlatformSettings> {
  const snap = await getDoc(settingsDoc())
  if (!snap.exists()) return defaultPlatformSettings
  return mapPlatformSettings(snap.id, snap.data())
}

export async function savePlatformSettings(settings: PlatformSettings, adminId: string): Promise<void> {
  await setDoc(settingsDoc(), {
    maintenanceMode: settings.maintenanceMode,
    maintenanceMessage: settings.maintenanceMessage,
    supportEmail: settings.supportEmail,
    supportPhone: settings.supportPhone,
    searchRadiiKm: settings.searchRadiiKm,
    candidateReviewSeconds: settings.candidateReviewSeconds,
    minSupportedAppVersion: settings.minSupportedAppVersion,
    updatedAt: serverTimestamp(),
    updatedByAdminId: adminId,
  })
  await logAuditEvent({
    action: 'platform_settings.update',
    targetType: 'platformSettings',
    targetId: 'config',
    details: {
      maintenanceMode: settings.maintenanceMode,
      minSupportedAppVersion: settings.minSupportedAppVersion,
    },
  })
}
